pkg/client/connector/service.go

```diff
@@ -237,11 +237,13 @@ func (s *service) UserNotifications(_ *empty.Empty, stream rpc.Connector_UserNot
 
 func (s *service) Login(ctx context.Context, _ *empty.Empty) (*rpc.LoginResult, error) {
 	ctx = s.callCtx(ctx, "Login")
-	resultCode, err := auth.EnsureLoggedIn(ctx, s.loginExecutor)
-	if err != nil {
+	if _, err := s.loginExecutor.GetToken(ctx); err == nil {
```

Comment:
Ah okay I was confused about where the refresh was actually happening but found it happens here: https://github.com/telepresenceio/telepresence/blob/release/v2/pkg/client/connector/auth/refresh.go#L52 

and this PR makes it so we actually check the token + refresh if necessary instead of just checking for the presence of `tokens.json`